packages/opentelemetry-test-utils/src/test-fixtures.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Utilities for running test files out-of-process. See `runTestFixture()` below.
+
+import * as assert from 'assert';
+import { execFile } from 'child_process';
+import { EventEmitter } from 'stream';
+import { IncomingMessage, ServerResponse, createServer } from 'http';
+import type { AddressInfo } from 'net';
+import { URL } from 'url';
+import { createGunzip } from 'zlib';
+
+import {
+  IInstrumentationScope,
+  IResource,
+  ISpan,
+} from '@opentelemetry/otlp-transformer';
+import { NodeSDK, tracing } from '@opentelemetry/sdk-node';
+import type { InstrumentationOption } from '@opentelemetry/instrumentation';
+
+/**
+ * A utility for scripts that will be run with `runTestFixture()` to create an
+ * appropriately configured NodeSDK.
+ *
+ * Typically, when run via `runTestFixture`, OTEL_EXPORTER_OTLP_ENDPOINT will be
+ * set to export to a test collector. When that envvar is not set, this falls
+ * back to exporting to the console for dev convenience.
+ */
+export function createTestNodeSdk(opts: {
+  serviceName?: string;
+  instrumentations: InstrumentationOption[];
+}) {
+  const spanProcessor = process.env.OTEL_EXPORTER_OTLP_ENDPOINT
+    ? undefined
+    : new tracing.SimpleSpanProcessor(new tracing.ConsoleSpanExporter());
+  const sdk = new NodeSDK({
+    serviceName: opts.serviceName || 'test-service',
+    spanProcessor,
+    instrumentations: opts.instrumentations,
+  });
+  return sdk;
+}
+
+// TestSpan is an OTLP span plus references to `resource` and
+// `instrumentationScope` that are shared between multiple spans in the
+// protocol.
+export type TestSpan = ISpan & {
+  resource: IResource;
+  instrumentationScope: IInstrumentationScope;
+};
+
+/**
+ * A minimal HTTP server that can act as an OTLP HTTP/JSON protocol collector.
+ * It stores the received data for later test assertions.
+ *
+ * Limitations: This only supports traces so far, no metrics or logs.
+ * There is little error checking here; we are assuming valid OTLP.
```

Comment:
I'd say this is likely fine. We can expand this later if necessary.
Little error checking also sounds okay to me, since it's for testing.